nbresuse/prometheus.py

```diff
@@ -1,56 +1,69 @@
+from typing import Optional
+
 from notebook.notebookapp import NotebookApp
 from prometheus_client import Gauge
-from tornado import gen
-from typing import Optional
 
-from nbresuse.metrics import cpu_metrics
-from nbresuse.metrics import CPUMetrics
-from nbresuse.metrics import memory_metrics
-from nbresuse.metrics import MemoryMetrics
+from nbresuse.metrics import PSUtilMetricsLoader
 
 try:
     # Traitlets >= 4.3.3
     from traitlets import Callable
 except ImportError:
     from .utils import Callable
 
-TOTAL_MEMORY_USAGE = Gauge("total_memory_usage", "counter for total memory usage", [])
 
-MAX_MEMORY_USAGE = Gauge("max_memory_usage", "counter for max memory usage", [])
-
-TOTAL_CPU_USAGE = Gauge("total_cpu_usage", "counter for total cpu usage", [])
+class PrometheusHandler(Callable):
+    def __init__(self, metricsloader: PSUtilMetricsLoader):
+        super().__init__()
+        self.metricsloader = metricsloader
+        self.config = metricsloader.config
+        self.session_manager = metricsloader.nbapp.session_manager
 
-MAX_CPU_USAGE = Gauge("max_cpu_usage", "counter for max cpu usage", [])
+        self.TOTAL_MEMORY_USAGE = Gauge(
```

Comment:
I would eventually like to replace this manually specified attributes and attribute names (one for each gauge) with automatically generated ones based on the user's specified configuration, using for loops similar to what was done [here](https://github.com/jupyter/nbviewer/blob/dda310767d9c209a985abadddfad389c404d12be/nbviewer/providers/base.py#L63) in NBViewer.

https://github.com/jupyter/nbviewer/blob/dda310767d9c209a985abadddfad389c404d12be/nbviewer/providers/base.py#L63

I think computing the value for the third argument of `setattr` will be more complicated than just getting an entry from a dictionary based on its key, but it should still be regular enough that it could be encompassed in a function, I think, allowing the call to `setattr` to still be legible.